src/TSHelper.ts

```diff
@@ -220,12 +222,50 @@ export class TSHelper {
         return [false, null];
     }
 
+    public static isNonCompoundPrefixUnaryOperator(node: ts.PrefixUnaryExpression): boolean {
+        return node.operator === ts.SyntaxKind.ExclamationToken
+               || node.operator === ts.SyntaxKind.MinusToken
+               || node.operator === ts.SyntaxKind.PlusToken
+               || node.operator === ts.SyntaxKind.TildeToken;
+    }
+
+    public static getUnaryCompoundAssignmentOperator(node: ts.PrefixUnaryExpression | ts.PostfixUnaryExpression)
+        : ts.BinaryOperator {
+        switch (node.operator) {
+            case ts.SyntaxKind.PlusPlusToken:
+                return ts.SyntaxKind.PlusToken;
+            case ts.SyntaxKind.MinusMinusToken:
+                return ts.SyntaxKind.MinusToken;
+            default:
+                throw TSTLErrors.UnsupportedKind(
+                    `unary ${ts.isPrefixUnaryExpression(node) ? "prefix" : "postfix"} operator`, node.operator, node);
+        }
+    }
+
     public static isExpressionStatement(node: ts.Expression): boolean {
         return node.parent === undefined
             || ts.isExpressionStatement(node.parent)
             || ts.isForStatement(node.parent);
     }
 
+    public static createLuaConcatExpression(left: ts.Expression, right: ts.Expression): LuaConcatExpression {
```

Comment:
I think @lolleko made a TransformHelper.ts for functions like these.

Reply:
Totally missed it! I'll move these over.